server/src/main/java/org/candlepin/model/EntitlementCurator.java

```diff
@@ -987,4 +924,132 @@ public Entitlement findUpstreamEntitlementForStack(Consumer consumer, String sta
 
         return (Entitlement) activeNowQuery.uniqueResult();
     }
+
+    /**
+     * Defines the queries used to find the collection of modifier entitlements of
+     * a given set of entitlements.
+     */
+    private class DependentEntitlementQuery {
```

Comment:
Why are we encapsulating this into a class? Nothing here seems particularly reusable, nor does it hold/manage any state for us.